### ISSUE TITLE
Unable run faust in windows 10

### ISSUE THREAD

**Issue by ST2Labs** [robinhood/faust#240]
## Actual behavior

I can't run hello-world example

## Full traceback

```pytb
(kafka) PS C:\Code\kafka> python .\test.py --no_color -l info -A test worker
┌ƒaµS† v1.4.1──────────────────────────────────────────┐
│ id        │ hello-world                              │
│ transport │ [URL('kafka://kafka.st2labs.com:80')]    │
│ store     │ memory:                                  │
│ web       │ http://wailabs:6066                      │
│ log       │ -stderr- (info)                          │
│ pid       │ 104220                                   │
│ hostname  │ WAILabs                                  │
│ platform  │ CPython 3.7.0 (Windows AMD64)            │
│ drivers   │ aiokafka=0.4.19 aiohttp=3.4.4            │
│ datadir   │ C:\Code\kafka\hello-world-data           │
│ appdir    │ C:\Code\kafka\hello-world-data\v1        │
└───────────┴──────────────────────────────────────────┘
CANNOT SETUP LOGGING: ValueError("Unable to configure formatter 'colored'") from   File ".\test.py", line 19, in <module>
    app.main()
  File "C:\Users\Julian J Gonzalez\envs\kafka\lib\site-packages\faust\app\base.py", line 579, in main
    cli(app=self)
  File "C:\Users\Julian J Gonzalez\envs\kafka\lib\site-packages\click\core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "C:\Users\Julian J Gonzalez\envs\kafka\lib\site-packages\click\core.py", line 697, in main
    rv = self.invoke(ctx)
  File "C:\Users\Julian J Gonzalez\envs\kafka\lib\site-packages\click\core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "C:\Users\Julian J Gonzalez\envs\kafka\lib\site-packages\click\core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "C:\Users\Julian J Gonzalez\envs\kafka\lib\site-packages\click\core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "C:\Users\Julian J Gonzalez\envs\kafka\lib\site-packages\click\decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "C:\Users\Julian J Gonzalez\envs\kafka\lib\site-packages\faust\cli\base.py", line 463, in _inner
    return cmd()
  File "C:\Users\Julian J Gonzalez\envs\kafka\lib\site-packages\faust\cli\base.py", line 538, in __call__
    self.run_using_worker(*args, **kwargs)
  File "C:\Users\Julian J Gonzalez\envs\kafka\lib\site-packages\faust\cli\base.py", line 547, in run_using_worker
    return worker.execute_from_commandline()
  File "C:\Users\Julian J Gonzalez\envs\kafka\lib\site-packages\mode\worker.py", line 219, in execute_from_commandline
    self.loop.run_until_complete(self._starting_fut)
  File "c:\users\julian j gonzalez\appdata\local\programs\python\python37-32\Lib\asyncio\base_events.py", line 555, in run_until_complete
    self.run_forever()
  File "c:\users\julian j gonzalez\appdata\local\programs\python\python37-32\Lib\asyncio\base_events.py", line 523, in run_forever
    self._run_once()
  File "c:\users\julian j gonzalez\appdata\local\programs\python\python37-32\Lib\asyncio\base_events.py", line 1758, in _run_once
    handle._run()
  File "c:\users\julian j gonzalez\appdata\local\programs\python\python37-32\Lib\asyncio\events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "C:\Users\Julian J Gonzalez\envs\kafka\lib\site-packages\mode\worker.py", line 280, in start
    await super().start()
  File "C:\Users\Julian J Gonzalez\envs\kafka\lib\site-packages\mode\services.py", line 706, in start
    await self._default_start()
  File "C:\Users\Julian J Gonzalez\envs\kafka\lib\site-packages\mode\services.py", line 713, in _default_start
    await self._actually_start()
  File "C:\Users\Julian J Gonzalez\envs\kafka\lib\site-packages\mode\services.py", line 720, in _actually_start
    await self.on_first_start()
  File "C:\Users\Julian J Gonzalez\envs\kafka\lib\site-packages\faust\worker.py", line 242, in on_first_start
    await self.default_on_first_start()
  File "C:\Users\Julian J Gonzalez\envs\kafka\lib\site-packages\mode\worker.py", line 132, in default_on_first_start
    self._setup_logging()
  File "C:\Users\Julian J Gonzalez\envs\kafka\lib\site-packages\mode\worker.py", line 154, in _setup_logging
    traceback.print_stack(file=sys.stderr)
stopping➢
```

# Versions

* Python version 3.7
* Faust version 1.4.1
* Operating system Windows 10
* Kafka version 0.10.1
* RocksDB version (if applicable)

**Comment by chrisseto** [comment]
I'm on macOS, otherwise have all the same versions and am seeing the same issue.

It looks to be coming from `logging.config.dictConfig` throwing a `KeyError('stream')` when loading the [`colored` config](https://github.com/ask/mode/blob/d59b8632f1643a69765cb6c4e988675ee2a67c60/mode/utils/logging.py#L73).

Haven't figured out the appropriate fix just yet.

**Comment by Fahreeve** [comment]
The temporary decision is installing `colorlog==3.2.0` instead of `colorlog==4.0.1`
Because default value was removed:
old: https://github.com/borntyping/python-colorlog/blob/v3.2.0/colorlog/colorlog.py#L200
NEW: https://github.com/borntyping/python-colorlog/blob/v4.0.1/colorlog/colorlog.py#L200

**Comment by ST2Labs** [comment]
Greats! ...

Do `pip install colorlog==3.2.0 --upgrade` 

And now running !

Thanks ! 👍

**Comment by europelee** [comment]
It may be better that use fixed version in file default.txt at requirements directory, so faust can be friendly to users.

**Comment by ask** [comment]
Thanks for reporting! This is fixed in 1.4.2

@europelee,
libraries cannot use fixed/pinned requirements, this will only lead to headache.

consider this scenario:

```
YourApp
    faust==1.4.0
        colorlog==3.2.0
    otherlib==0.3.1
        colorlog==3.0.9
```
This will make pip complain with an error since both libraries are asking for differing versions.

What we can do is depend on a version range like:

```
faust
    colorlog>=3.2.0,<4.0
```

Sadly this will break if they introduce a backwards incompatible change to a minor version bump :(
